libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
+		defer brq.prefetchMtx.Unlock()
+		shutdownCh = brq.prefetcher.Shutdown()
+	}()
 	select {
-	case <-brq.prefetcher.Shutdown():
+	case <-shutdownCh:
 	case <-ctx.Done():
+		err = ctx.Err()
 	}
 	if enable {
-		brq.prefetcher = newBlockPrefetcher(brq)
+		brq.prefetcher = newBlockPrefetcher(brq, brq.config)
```

Comment:
Seems safer to set this under a lock too, though it's probably ok as is since Go pointer updates should be atomic.